Makefile.include

```diff
@@ -514,6 +514,9 @@ ifneq (, $(filter all, $(if $(MAKECMDGOALS), $(MAKECMDGOALS), all)))
     EXPECT_ERRORS := 1
   endif
 
+  # turn provided but blacklisted features into required "!<feature>"
+  FEATURES_REQUIRED += $(addprefix !,$(sort $(filter $(FEATURES_PROVIDED), $(FEATURES_BLACKLIST))))
```

Comment:
I would prefer adding another variable name and/or other if case for this check.
These sections are hard enough to read, so adding an explicit name simplify this.

Reply:
You mean, instead of directly adding the blacklisted but available features to FEATURES_REQUIRED, create an intermediate FEATURES_BLACKLISTED and use that in the if below?

I thought about that, but decided to rather go with a nice comment... I think ```warning: feature "!arch_16bit" missing, expect errors``` pretty much explains what is happening, and extra variables clutter code.

Do you insist?

Reply:
addressed